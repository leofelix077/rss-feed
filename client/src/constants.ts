export enum SupportedLocale {
  English = "en",
  Portuguese = "pt",
}

export enum SupportedTemperature {
  Celsius = "C",
  Fahrenheit = "F",
}

export const LOCALE_KEY = "locale";

export const TEMPERATURE_KEY = "temperature";

export const WEATHER_API = "http://api.openweathermap.org/data/2.5/";
