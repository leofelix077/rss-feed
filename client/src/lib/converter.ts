import {
  SupportedTemperature,
  TEMPERATURE_KEY,
  ABSOLUTE_ZERO,
} from "../constants";
import i18next from "i18next";

export const convertTemperature = (
  degrees: number,
  temperature: SupportedTemperature
): number => {
  switch (temperature) {
    case SupportedTemperature.Celsius:
      return Math.round(degrees + ABSOLUTE_ZERO);
    case SupportedTemperature.Fahrenheit:
      return Math.round(((degrees + ABSOLUTE_ZERO) * 9) / 5 + 32);
    default:
      localStorage.removeItem(TEMPERATURE_KEY);
      throw new Error(i18next.t("errors:temperature-unsupported"));
  }
};
