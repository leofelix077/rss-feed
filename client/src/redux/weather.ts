/* eslint-disable no-console */
/* eslint-disable no-param-reassign */
import { takeLatest, put, call } from "redux-saga/effects";
import { SagaIterator } from "redux-saga";
import produce from "immer";
import { transport } from "../lib/transport";
import { createWeatherApiCall } from "../lib/api";

// types
export const GET_WEATHER = {
  REQUEST: "GET_WEATHER.REQUEST",
  SUCCESS: "GET_WEATHER.SUCCESS",
  ERROR: "GET_WEATHER.ERROR",
};

interface WeatherLocation {
  city: string;
  isoCountryCode: string;
}

export interface WeatherDescription {
  id: number;
  main: string;
  description: string;
  icon: string;
}

export interface WeatherDataPoint {
  dt: number;
  main: {
    temp: number;
    feels_like: number;
    temp_min: number;
    temp_max: number;
    pressure: number;
    sea_level: number;
    grnd_level: number;
    humidity: number;
    temp_kf: number;
  };
  weather: WeatherDescription[];
  clouds: {
    all: number;
  };
  wind: {
    speed: number;
    deg: number;
  };
  sys: {
    pod: string;
  };
  dt_txt: string;
}

export interface WeatherCityInformation {
  id: number;
  name: string;
  coord: { lat: number; lon: number };
  country: string;
  population: number;
  timezone: number;
  sunrise: number;
  sunset: number;
}

export interface WeatherData {
  cod: string;
  message: number;
  cnt: number;
  list: WeatherDataPoint[];
  city: WeatherCityInformation;
}

// actions
interface GetCurrentWeatherRequest extends WeatherLocation {
  type: typeof GET_WEATHER.REQUEST;
}

export const getCurrentWeatherRequest = (
  city: string,
  isoCountryCode: string
): GetCurrentWeatherRequest => ({
  type: GET_WEATHER.REQUEST,
  city,
  isoCountryCode,
});

interface GetCurrentWeatherSuccess {
  type: typeof GET_WEATHER.SUCCESS;
  weatherForecast: WeatherData;
}

export const getCurrentWeatherSuccess = (
  weatherForecast: WeatherData
): GetCurrentWeatherSuccess => ({
  type: GET_WEATHER.SUCCESS,
  weatherForecast,
});

interface GetCurrentWeatherError {
  type: typeof GET_WEATHER.ERROR;
  error: string;
}

export const getCurrentWeatherError = (
  error: string
): GetCurrentWeatherError => ({
  type: GET_WEATHER.SUCCESS,
  error,
});

// reducer
interface WeatherForecast {
  weatherForecast: WeatherData | null;
  request: {
    processing: boolean;
    success: boolean;
    error: string | null;
  };
}

export const initialState: WeatherForecast = {
  weatherForecast: null,
  request: {
    processing: false,
    success: false,
    error: null,
  },
};

export function weatherReducer(
  state = initialState,
  action: any
): WeatherForecast {
  switch (action.type) {
    case GET_WEATHER.REQUEST:
      return produce(state, (newState) => {
        newState.request.processing = true;
      });
    case GET_WEATHER.SUCCESS:
      return produce(state, (newState) => {
        newState.weatherForecast = action.weatherForecast;
        newState.request.success = true;
        newState.request.processing = false;
      });
    case GET_WEATHER.ERROR:
      return produce(state, (newState) => {
        newState.request.error = action.error;
        newState.request.success = false;
        newState.request.processing = action.false;
      });
    default:
      return state;
  }
}

// saga

function* weatherApiSaga(action: WeatherLocation): any {
  const { city, isoCountryCode } = action;

  try {
    const response: WeatherData = yield call(transport, {
      url: createWeatherApiCall(city, isoCountryCode),
    });
    console.log(response);
    yield put(getCurrentWeatherSuccess(response));
  } catch (error) {
    console.log(error);
    yield put(getCurrentWeatherError(error.message));
  }
}

export function* weatherApi(): SagaIterator {
  yield takeLatest(GET_WEATHER.REQUEST as any, weatherApiSaga);
}
