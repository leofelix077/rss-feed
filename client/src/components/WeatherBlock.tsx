import React from "react";
import { makeStyles, Typography } from "@material-ui/core";
import { WeatherDataPoint } from "../redux/weather";
import PropTypes from "prop-types";
import { convertTemperature } from "../lib/converter";
import { useSelector } from "react-redux";
import { RootState } from "../redux/rootReducer";
import { useTranslation } from "react-i18next";
import moment from "moment";
import clsx from "clsx";

interface WeatherBlockProps {
  data: WeatherDataPoint[];
  day: number;
  isSelected: boolean;
  utcOffset: number;
}

const useStyles = makeStyles((theme) => ({
  container: {
    display: "flex",
    flexDirection: "column",
    alignItems: "center",
    justifyContent: "center",
    padding: theme.spacing(1),
    width: "100%",
    margin: theme.spacing(2),
    borderRadius: 4,
    boxShadow: "2px 2px rgba(0,0,0,0.2)",
    backgroundColor: "#141414",
  },
  temperature: {
    color: "white",
    fontSize: "6vw",
  },
  additionalInfoText: {
    color: "white",
    fontSize: "2vw",
  },
  additionalInfoHeader: {
    color: "white",
    fontSize: "1vw",
  },
  additionalInfoContainer: {
    display: "flex",
    flexDirection: "column",
    alignItems: "center",
    padding: theme.spacing(1),
  },
  additionalInfo: {
    flexDirection: "row",
    justifyContent: "space-around",
    display: "flex",
    padding: theme.spacing(1),
    width: "100%",
  },
  feelsLikeContainer: {
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "center",
  },
  feelsLikeHeader: {
    fontSize: "1vw",
  },
  feelsLike: {
    fontSize: "2vw",
  },
  containerSelected: {
    boxShadow: "2px 2px rgba(128,128,128,0.2)",
  },
  date: {
    fontSize: "2vw",
  },
}));

const WeatherBlock: React.FC<WeatherBlockProps> = ({
  data,
  day,
  isSelected,
  utcOffset,
}): ReturnType<React.FC<WeatherBlockProps>> => {
  const classes = useStyles();

  const { t } = useTranslation("weather");

  const locale = useSelector(
    (state: RootState) => state.localeDetector.currentLocale
  );
  const preferredTemperature = useSelector(
    (state: RootState) => state.temperature.temperature
  );

  const temps = data.reduce(
    (acc, dataPoint) => {
      acc.avg += convertTemperature(dataPoint.main.temp, preferredTemperature);
      acc.humidity += dataPoint.main.humidity;
      acc.pressure += dataPoint.main.pressure;
      acc.feelsLike += convertTemperature(
        dataPoint.main.feels_like,
        preferredTemperature
      );
      return acc;
    },
    {
      avg: 0,
      humidity: 0,
      pressure: 0,
      feelsLike: 0,
    }
  );

  const finalTemperature = Math.floor(temps.avg / data.length);
  const finalHumidity = Math.floor(temps.humidity / data.length);
  const finalPressure = Math.floor(temps.pressure / data.length);
  const finalFeelsLike = Math.floor(temps.feelsLike / data.length);

  return (
    <div
      className={clsx({
        [classes.container]: true,
        [classes.containerSelected]: isSelected,
      })}
    >
      <Typography className={classes.date}>
        {moment
          .unix(day)
          .utcOffset(utcOffset / 60)
          .locale(locale)
          .format("L")}
      </Typography>
      <Typography className={classes.temperature}>
        {finalTemperature} °{preferredTemperature}
      </Typography>
      <div>
        <Typography className={classes.feelsLikeHeader}>
          {t("feelsLike")}
        </Typography>
        <Typography className={classes.feelsLike}>
          {finalFeelsLike} °{preferredTemperature}
        </Typography>
      </div>
      <div className={classes.additionalInfo}>
        <div className={classes.additionalInfoContainer}>
          <Typography className={classes.additionalInfoHeader}>
            {t("humidity")}
          </Typography>
          <Typography className={classes.additionalInfoText}>
            {finalHumidity}%
          </Typography>
        </div>
        <div className={classes.additionalInfoContainer}>
          <Typography className={classes.additionalInfoHeader}>
            {t("pressure")}
          </Typography>
          <Typography className={classes.additionalInfoText}>
            {finalPressure} hpa
          </Typography>
        </div>
      </div>
    </div>
  );
};

WeatherBlock.propTypes = {
  data: PropTypes.array.isRequired,
  day: PropTypes.number.isRequired,
  isSelected: PropTypes.bool.isRequired,
  utcOffset: PropTypes.number.isRequired,
};

export default WeatherBlock;
