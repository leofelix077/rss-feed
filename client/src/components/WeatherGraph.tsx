import React from "react";
import { WeatherDataPoint } from "../redux/weather";
import PropTypes from "prop-types";
import { makeStyles } from "@material-ui/core";
import { Bar } from "react-chartjs-2";
import { useTranslation } from "react-i18next";
import moment from "moment";
import { convertTemperature } from "../lib/converter";
import { useSelector } from "react-redux";
import { RootState } from "../redux/rootReducer";

interface WeatherGraphProps {
  data: WeatherDataPoint[];
  utcOffset: number;
}

const useStyles = makeStyles((theme) => ({
  container: {
    padding: theme.spacing(2),
    width: "60vw",
    display: "flex",
    position: "relative",
    justifyContent: "center",
    height: "30vh",
  },
  root: {
    width: "100%",
    display: "flex",
    alignItems: "center",
    justifyContent: "center",
  },
}));

export const WeatherGraph: React.FC<WeatherGraphProps> = ({
  data,
  utcOffset,
}): ReturnType<React.FC<WeatherGraphProps>> => {
  const classes = useStyles();

  const { t } = useTranslation("weather");

  const preferredTemperature = useSelector(
    (state: RootState) => state.temperature.temperature
  );

  if (!data) {
    return null;
  }

  const labels = data.map((dataPoint) =>
    moment
      .unix(dataPoint.dt)
      .utcOffset(utcOffset / 60)
      .format("HH:mm")
  );

  return (
    <div className={classes.root}>
      <div className={classes.container}>
        <Bar
          data={{
            labels,
            datasets: [
              {
                label: t("graphLabel"),
                backgroundColor: "rgba(0,99,0,0.2)",
                borderColor: "rgba(0,99,0,1)",
                borderWidth: 1,
                hoverBackgroundColor: "rgba(0,99,0,0.4)",
                hoverBorderColor: "rgba(0,99,0,1)",
                data: data.map((dataPoint) =>
                  convertTemperature(dataPoint.main.temp, preferredTemperature)
                ),
              },
            ],
          }}
          options={{
            maintainAspectRatio: true,
            scales: {
              yAxes: [
                {
                  ticks: {
                    beginAtZero: true,
                  },
                },
              ],
            },
          }}
        />
      </div>
    </div>
  );
};

WeatherGraph.propTypes = {
  data: PropTypes.array.isRequired,
  utcOffset: PropTypes.number.isRequired,
};
