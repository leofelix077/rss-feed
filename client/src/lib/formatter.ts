import groupBy from "lodash/groupBy";
import { Dictionary } from "lodash";
import { WeatherDataPoint, WeatherData } from "../redux/weather";
import moment from "moment";

export const formatSelectedDay = (
  selectedDay: number,
  timezoneOffset: number
): number => {
  return moment
    .unix(selectedDay)
    .utcOffset(timezoneOffset / 60)
    .startOf("day")
    .unix();
};

export const formatWeatherData = (
  weatherData: WeatherData
): Dictionary<WeatherDataPoint[]> => {
  const formattedWeatherData = groupBy(
    weatherData.list,
    (dataPoint: WeatherDataPoint) =>
      formatSelectedDay(dataPoint.dt, weatherData.city.timezone)
  );

  return formattedWeatherData;
};
