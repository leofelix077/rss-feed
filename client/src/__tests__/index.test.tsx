import React from "react";
import { mount } from "enzyme";
import { describe, expect, it } from "@jest/globals";
import WeatherBlock from "../components/WeatherBlock";
import { getRootStateMock } from "../testData/data";
import { getRootStateMockDifferentLocale } from "../testData/dataDifferentLocale";
import { WeatherData } from "../redux/weather";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import "jsdom-global/register";
import mockdate from "mockdate";
import { WeatherGraph } from "../components/WeatherGraph";
import { formatWeatherData, formatSelectedDay } from "../lib/formatter";
import moment from "moment";
import { convertTemperature } from "../lib/converter";
import { SupportedTemperature, ABSOLUTE_ZERO } from "../constants";

const mockStore = configureMockStore();

describe("Forecast for locale, day and utc offset", () => {
  it("renders WeatherBlock correctly", () => {
    mockdate.set(1593118700000);
    const store = getRootStateMock();

    const weatherData = store.weather.weatherForecast as WeatherData;

    const formattedWeatherData = formatWeatherData(weatherData);

    const selectedDay = formatSelectedDay(
      weatherData.list[0].dt,
      weatherData.city.timezone
    );

    const storeProvider = mockStore(store);

    const wrapper = mount(
      <Provider store={storeProvider}>
        <WeatherBlock
          data={formattedWeatherData[selectedDay]}
          day={selectedDay}
          isSelected
          utcOffset={weatherData.city.timezone}
        />
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();

    const element = wrapper.find("p#temperature-text");
    expect(element.length).toBe(1);
    expect(element.children).toBeDefined();
    expect((element as any).props().children.join("")).toBe("18 Â°C");
  });

  it("renders WeatherGraph correctly for differentLocale", () => {
    mockdate.set(1593118700000);
    const store = getRootStateMockDifferentLocale();

    const weatherData = store.weather.weatherForecast as WeatherData;

    const formattedWeatherData = formatWeatherData(weatherData);

    const selectedDay = formatSelectedDay(
      weatherData.list[0].dt,
      weatherData.city.timezone
    );

    const storeProvider = mockStore(store);

    const wrapper = mount(
      <Provider store={storeProvider}>
        <WeatherGraph
          data={formattedWeatherData[selectedDay]}
          utcOffset={weatherData.city.timezone}
        />
      </Provider>
    );
    expect(wrapper.props()).toMatchSnapshot();
    const element = wrapper.find("Bar").props();
    const elementData: string[] = (element as any).data.labels;
    expect(elementData.slice(1)).not.toContain("00:00");
    expect(elementData).toMatchSnapshot();
    wrapper.unmount();

    const wrapper2 = mount(
      <Provider store={storeProvider}>
        <WeatherGraph
          data={
            formattedWeatherData[moment.unix(selectedDay).add(1, "day").unix()]
          }
          utcOffset={weatherData.city.timezone}
        />
      </Provider>
    );
    expect(wrapper2.props()).toMatchSnapshot();
    const element2 = wrapper2.find("Bar").props();
    const element2Data: string[] = (element2 as any).data.labels;

    expect(element2Data.slice(1)).not.toContain("00:00");
    expect(element2Data[0]).toBe("00:00");
    expect(element2Data[element2Data.length - 1]).toBe("21:00");
    expect(element2Data).toMatchSnapshot();
    wrapper2.unmount();
  });

  it("should convert temperature correctly", () => {
    const fahrenheitZeroTemperature = convertTemperature(
      ABSOLUTE_ZERO * -1,
      SupportedTemperature.Fahrenheit
    );
    const celsiusZeroTemperature = convertTemperature(
      ABSOLUTE_ZERO * -1,
      SupportedTemperature.Celsius
    );

    expect(fahrenheitZeroTemperature).toEqual(32);
    expect(celsiusZeroTemperature).toEqual(0);

    const fahrenheitBoilWaterTemperature = convertTemperature(
      ABSOLUTE_ZERO * -1 + 100,
      SupportedTemperature.Fahrenheit
    );
    const celsiusBoilWaterTemperature = convertTemperature(
      ABSOLUTE_ZERO * -1 + 100,
      SupportedTemperature.Celsius
    );

    expect(fahrenheitBoilWaterTemperature).toEqual(212);
    expect(celsiusBoilWaterTemperature).toEqual(100);
  });
});
