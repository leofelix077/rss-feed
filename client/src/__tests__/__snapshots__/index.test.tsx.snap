// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render each weather block with correct temperature renders WeatherBlock correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <WeatherBlock
    data={
      Array [
        Object {
          "clouds": Object {
            "all": 14,
          },
          "dt": 1593118800,
          "dt_txt": "2020-06-25 21:00:00",
          "main": Object {
            "feels_like": 291.01,
            "grnd_level": 958,
            "humidity": 75,
            "pressure": 1018,
            "sea_level": 1018,
            "temp": 290.7,
            "temp_kf": 3.18,
            "temp_max": 290.7,
            "temp_min": 287.52,
          },
          "sys": Object {
            "pod": "n",
          },
          "weather": Array [
            Object {
              "description": "few clouds",
              "icon": "02n",
              "id": 801,
              "main": "Clouds",
            },
          ],
          "wind": Object {
            "deg": 132,
            "speed": 0.92,
          },
        },
      ]
    }
    day={0}
    isSelected={true}
    utcOffset={0}
  >
    <div
      className="makeStyles-container-1 makeStyles-containerSelected-10"
    >
      <WithStyles(ForwardRef(Typography))
        className="makeStyles-date-11"
      >
        <ForwardRef(Typography)
          className="makeStyles-date-11"
          classes={
            Object {
              "alignCenter": "MuiTypography-alignCenter",
              "alignJustify": "MuiTypography-alignJustify",
              "alignLeft": "MuiTypography-alignLeft",
              "alignRight": "MuiTypography-alignRight",
              "body1": "MuiTypography-body1",
              "body2": "MuiTypography-body2",
              "button": "MuiTypography-button",
              "caption": "MuiTypography-caption",
              "colorError": "MuiTypography-colorError",
              "colorInherit": "MuiTypography-colorInherit",
              "colorPrimary": "MuiTypography-colorPrimary",
              "colorSecondary": "MuiTypography-colorSecondary",
              "colorTextPrimary": "MuiTypography-colorTextPrimary",
              "colorTextSecondary": "MuiTypography-colorTextSecondary",
              "displayBlock": "MuiTypography-displayBlock",
              "displayInline": "MuiTypography-displayInline",
              "gutterBottom": "MuiTypography-gutterBottom",
              "h1": "MuiTypography-h1",
              "h2": "MuiTypography-h2",
              "h3": "MuiTypography-h3",
              "h4": "MuiTypography-h4",
              "h5": "MuiTypography-h5",
              "h6": "MuiTypography-h6",
              "noWrap": "MuiTypography-noWrap",
              "overline": "MuiTypography-overline",
              "paragraph": "MuiTypography-paragraph",
              "root": "MuiTypography-root",
              "srOnly": "MuiTypography-srOnly",
              "subtitle1": "MuiTypography-subtitle1",
              "subtitle2": "MuiTypography-subtitle2",
            }
          }
        >
          <p
            className="MuiTypography-root makeStyles-date-11 MuiTypography-body1"
          >
            01/01/1970
          </p>
        </ForwardRef(Typography)>
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Typography))
        className="makeStyles-temperature-2"
      >
        <ForwardRef(Typography)
          className="makeStyles-temperature-2"
          classes={
            Object {
              "alignCenter": "MuiTypography-alignCenter",
              "alignJustify": "MuiTypography-alignJustify",
              "alignLeft": "MuiTypography-alignLeft",
              "alignRight": "MuiTypography-alignRight",
              "body1": "MuiTypography-body1",
              "body2": "MuiTypography-body2",
              "button": "MuiTypography-button",
              "caption": "MuiTypography-caption",
              "colorError": "MuiTypography-colorError",
              "colorInherit": "MuiTypography-colorInherit",
              "colorPrimary": "MuiTypography-colorPrimary",
              "colorSecondary": "MuiTypography-colorSecondary",
              "colorTextPrimary": "MuiTypography-colorTextPrimary",
              "colorTextSecondary": "MuiTypography-colorTextSecondary",
              "displayBlock": "MuiTypography-displayBlock",
              "displayInline": "MuiTypography-displayInline",
              "gutterBottom": "MuiTypography-gutterBottom",
              "h1": "MuiTypography-h1",
              "h2": "MuiTypography-h2",
              "h3": "MuiTypography-h3",
              "h4": "MuiTypography-h4",
              "h5": "MuiTypography-h5",
              "h6": "MuiTypography-h6",
              "noWrap": "MuiTypography-noWrap",
              "overline": "MuiTypography-overline",
              "paragraph": "MuiTypography-paragraph",
              "root": "MuiTypography-root",
              "srOnly": "MuiTypography-srOnly",
              "subtitle1": "MuiTypography-subtitle1",
              "subtitle2": "MuiTypography-subtitle2",
            }
          }
        >
          <p
            className="MuiTypography-root makeStyles-temperature-2 MuiTypography-body1"
          >
            18
             °
            C
          </p>
        </ForwardRef(Typography)>
      </WithStyles(ForwardRef(Typography))>
      <div>
        <WithStyles(ForwardRef(Typography))
          className="makeStyles-feelsLikeHeader-8"
        >
          <ForwardRef(Typography)
            className="makeStyles-feelsLikeHeader-8"
            classes={
              Object {
                "alignCenter": "MuiTypography-alignCenter",
                "alignJustify": "MuiTypography-alignJustify",
                "alignLeft": "MuiTypography-alignLeft",
                "alignRight": "MuiTypography-alignRight",
                "body1": "MuiTypography-body1",
                "body2": "MuiTypography-body2",
                "button": "MuiTypography-button",
                "caption": "MuiTypography-caption",
                "colorError": "MuiTypography-colorError",
                "colorInherit": "MuiTypography-colorInherit",
                "colorPrimary": "MuiTypography-colorPrimary",
                "colorSecondary": "MuiTypography-colorSecondary",
                "colorTextPrimary": "MuiTypography-colorTextPrimary",
                "colorTextSecondary": "MuiTypography-colorTextSecondary",
                "displayBlock": "MuiTypography-displayBlock",
                "displayInline": "MuiTypography-displayInline",
                "gutterBottom": "MuiTypography-gutterBottom",
                "h1": "MuiTypography-h1",
                "h2": "MuiTypography-h2",
                "h3": "MuiTypography-h3",
                "h4": "MuiTypography-h4",
                "h5": "MuiTypography-h5",
                "h6": "MuiTypography-h6",
                "noWrap": "MuiTypography-noWrap",
                "overline": "MuiTypography-overline",
                "paragraph": "MuiTypography-paragraph",
                "root": "MuiTypography-root",
                "srOnly": "MuiTypography-srOnly",
                "subtitle1": "MuiTypography-subtitle1",
                "subtitle2": "MuiTypography-subtitle2",
              }
            }
          >
            <p
              className="MuiTypography-root makeStyles-feelsLikeHeader-8 MuiTypography-body1"
            >
              feelsLike
            </p>
          </ForwardRef(Typography)>
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          className="makeStyles-feelsLike-9"
        >
          <ForwardRef(Typography)
            className="makeStyles-feelsLike-9"
            classes={
              Object {
                "alignCenter": "MuiTypography-alignCenter",
                "alignJustify": "MuiTypography-alignJustify",
                "alignLeft": "MuiTypography-alignLeft",
                "alignRight": "MuiTypography-alignRight",
                "body1": "MuiTypography-body1",
                "body2": "MuiTypography-body2",
                "button": "MuiTypography-button",
                "caption": "MuiTypography-caption",
                "colorError": "MuiTypography-colorError",
                "colorInherit": "MuiTypography-colorInherit",
                "colorPrimary": "MuiTypography-colorPrimary",
                "colorSecondary": "MuiTypography-colorSecondary",
                "colorTextPrimary": "MuiTypography-colorTextPrimary",
                "colorTextSecondary": "MuiTypography-colorTextSecondary",
                "displayBlock": "MuiTypography-displayBlock",
                "displayInline": "MuiTypography-displayInline",
                "gutterBottom": "MuiTypography-gutterBottom",
                "h1": "MuiTypography-h1",
                "h2": "MuiTypography-h2",
                "h3": "MuiTypography-h3",
                "h4": "MuiTypography-h4",
                "h5": "MuiTypography-h5",
                "h6": "MuiTypography-h6",
                "noWrap": "MuiTypography-noWrap",
                "overline": "MuiTypography-overline",
                "paragraph": "MuiTypography-paragraph",
                "root": "MuiTypography-root",
                "srOnly": "MuiTypography-srOnly",
                "subtitle1": "MuiTypography-subtitle1",
                "subtitle2": "MuiTypography-subtitle2",
              }
            }
          >
            <p
              className="MuiTypography-root makeStyles-feelsLike-9 MuiTypography-body1"
            >
              18
               °
              C
            </p>
          </ForwardRef(Typography)>
        </WithStyles(ForwardRef(Typography))>
      </div>
      <div
        className="makeStyles-additionalInfo-6"
      >
        <div
          className="makeStyles-additionalInfoContainer-5"
        >
          <WithStyles(ForwardRef(Typography))
            className="makeStyles-additionalInfoHeader-4"
          >
            <ForwardRef(Typography)
              className="makeStyles-additionalInfoHeader-4"
              classes={
                Object {
                  "alignCenter": "MuiTypography-alignCenter",
                  "alignJustify": "MuiTypography-alignJustify",
                  "alignLeft": "MuiTypography-alignLeft",
                  "alignRight": "MuiTypography-alignRight",
                  "body1": "MuiTypography-body1",
                  "body2": "MuiTypography-body2",
                  "button": "MuiTypography-button",
                  "caption": "MuiTypography-caption",
                  "colorError": "MuiTypography-colorError",
                  "colorInherit": "MuiTypography-colorInherit",
                  "colorPrimary": "MuiTypography-colorPrimary",
                  "colorSecondary": "MuiTypography-colorSecondary",
                  "colorTextPrimary": "MuiTypography-colorTextPrimary",
                  "colorTextSecondary": "MuiTypography-colorTextSecondary",
                  "displayBlock": "MuiTypography-displayBlock",
                  "displayInline": "MuiTypography-displayInline",
                  "gutterBottom": "MuiTypography-gutterBottom",
                  "h1": "MuiTypography-h1",
                  "h2": "MuiTypography-h2",
                  "h3": "MuiTypography-h3",
                  "h4": "MuiTypography-h4",
                  "h5": "MuiTypography-h5",
                  "h6": "MuiTypography-h6",
                  "noWrap": "MuiTypography-noWrap",
                  "overline": "MuiTypography-overline",
                  "paragraph": "MuiTypography-paragraph",
                  "root": "MuiTypography-root",
                  "srOnly": "MuiTypography-srOnly",
                  "subtitle1": "MuiTypography-subtitle1",
                  "subtitle2": "MuiTypography-subtitle2",
                }
              }
            >
              <p
                className="MuiTypography-root makeStyles-additionalInfoHeader-4 MuiTypography-body1"
              >
                humidity
              </p>
            </ForwardRef(Typography)>
          </WithStyles(ForwardRef(Typography))>
          <WithStyles(ForwardRef(Typography))
            className="makeStyles-additionalInfoText-3"
          >
            <ForwardRef(Typography)
              className="makeStyles-additionalInfoText-3"
              classes={
                Object {
                  "alignCenter": "MuiTypography-alignCenter",
                  "alignJustify": "MuiTypography-alignJustify",
                  "alignLeft": "MuiTypography-alignLeft",
                  "alignRight": "MuiTypography-alignRight",
                  "body1": "MuiTypography-body1",
                  "body2": "MuiTypography-body2",
                  "button": "MuiTypography-button",
                  "caption": "MuiTypography-caption",
                  "colorError": "MuiTypography-colorError",
                  "colorInherit": "MuiTypography-colorInherit",
                  "colorPrimary": "MuiTypography-colorPrimary",
                  "colorSecondary": "MuiTypography-colorSecondary",
                  "colorTextPrimary": "MuiTypography-colorTextPrimary",
                  "colorTextSecondary": "MuiTypography-colorTextSecondary",
                  "displayBlock": "MuiTypography-displayBlock",
                  "displayInline": "MuiTypography-displayInline",
                  "gutterBottom": "MuiTypography-gutterBottom",
                  "h1": "MuiTypography-h1",
                  "h2": "MuiTypography-h2",
                  "h3": "MuiTypography-h3",
                  "h4": "MuiTypography-h4",
                  "h5": "MuiTypography-h5",
                  "h6": "MuiTypography-h6",
                  "noWrap": "MuiTypography-noWrap",
                  "overline": "MuiTypography-overline",
                  "paragraph": "MuiTypography-paragraph",
                  "root": "MuiTypography-root",
                  "srOnly": "MuiTypography-srOnly",
                  "subtitle1": "MuiTypography-subtitle1",
                  "subtitle2": "MuiTypography-subtitle2",
                }
              }
            >
              <p
                className="MuiTypography-root makeStyles-additionalInfoText-3 MuiTypography-body1"
              >
                75
                %
              </p>
            </ForwardRef(Typography)>
          </WithStyles(ForwardRef(Typography))>
        </div>
        <div
          className="makeStyles-additionalInfoContainer-5"
        >
          <WithStyles(ForwardRef(Typography))
            className="makeStyles-additionalInfoHeader-4"
          >
            <ForwardRef(Typography)
              className="makeStyles-additionalInfoHeader-4"
              classes={
                Object {
                  "alignCenter": "MuiTypography-alignCenter",
                  "alignJustify": "MuiTypography-alignJustify",
                  "alignLeft": "MuiTypography-alignLeft",
                  "alignRight": "MuiTypography-alignRight",
                  "body1": "MuiTypography-body1",
                  "body2": "MuiTypography-body2",
                  "button": "MuiTypography-button",
                  "caption": "MuiTypography-caption",
                  "colorError": "MuiTypography-colorError",
                  "colorInherit": "MuiTypography-colorInherit",
                  "colorPrimary": "MuiTypography-colorPrimary",
                  "colorSecondary": "MuiTypography-colorSecondary",
                  "colorTextPrimary": "MuiTypography-colorTextPrimary",
                  "colorTextSecondary": "MuiTypography-colorTextSecondary",
                  "displayBlock": "MuiTypography-displayBlock",
                  "displayInline": "MuiTypography-displayInline",
                  "gutterBottom": "MuiTypography-gutterBottom",
                  "h1": "MuiTypography-h1",
                  "h2": "MuiTypography-h2",
                  "h3": "MuiTypography-h3",
                  "h4": "MuiTypography-h4",
                  "h5": "MuiTypography-h5",
                  "h6": "MuiTypography-h6",
                  "noWrap": "MuiTypography-noWrap",
                  "overline": "MuiTypography-overline",
                  "paragraph": "MuiTypography-paragraph",
                  "root": "MuiTypography-root",
                  "srOnly": "MuiTypography-srOnly",
                  "subtitle1": "MuiTypography-subtitle1",
                  "subtitle2": "MuiTypography-subtitle2",
                }
              }
            >
              <p
                className="MuiTypography-root makeStyles-additionalInfoHeader-4 MuiTypography-body1"
              >
                pressure
              </p>
            </ForwardRef(Typography)>
          </WithStyles(ForwardRef(Typography))>
          <WithStyles(ForwardRef(Typography))
            className="makeStyles-additionalInfoText-3"
          >
            <ForwardRef(Typography)
              className="makeStyles-additionalInfoText-3"
              classes={
                Object {
                  "alignCenter": "MuiTypography-alignCenter",
                  "alignJustify": "MuiTypography-alignJustify",
                  "alignLeft": "MuiTypography-alignLeft",
                  "alignRight": "MuiTypography-alignRight",
                  "body1": "MuiTypography-body1",
                  "body2": "MuiTypography-body2",
                  "button": "MuiTypography-button",
                  "caption": "MuiTypography-caption",
                  "colorError": "MuiTypography-colorError",
                  "colorInherit": "MuiTypography-colorInherit",
                  "colorPrimary": "MuiTypography-colorPrimary",
                  "colorSecondary": "MuiTypography-colorSecondary",
                  "colorTextPrimary": "MuiTypography-colorTextPrimary",
                  "colorTextSecondary": "MuiTypography-colorTextSecondary",
                  "displayBlock": "MuiTypography-displayBlock",
                  "displayInline": "MuiTypography-displayInline",
                  "gutterBottom": "MuiTypography-gutterBottom",
                  "h1": "MuiTypography-h1",
                  "h2": "MuiTypography-h2",
                  "h3": "MuiTypography-h3",
                  "h4": "MuiTypography-h4",
                  "h5": "MuiTypography-h5",
                  "h6": "MuiTypography-h6",
                  "noWrap": "MuiTypography-noWrap",
                  "overline": "MuiTypography-overline",
                  "paragraph": "MuiTypography-paragraph",
                  "root": "MuiTypography-root",
                  "srOnly": "MuiTypography-srOnly",
                  "subtitle1": "MuiTypography-subtitle1",
                  "subtitle2": "MuiTypography-subtitle2",
                }
              }
            >
              <p
                className="MuiTypography-root makeStyles-additionalInfoText-3 MuiTypography-body1"
              >
                1018
                 hpa
              </p>
            </ForwardRef(Typography)>
          </WithStyles(ForwardRef(Typography))>
        </div>
      </div>
    </div>
  </WeatherBlock>
</Provider>
`;
